component/grant-type/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/TokenExchangeGrantHandler.java

```diff
@@ -0,0 +1,326 @@
+    private String requested_token_type = TokenExchangeConstants.JWT_TOKEN_TYPE;
+
+    /**
+     * Initialize the TokenExchangeGrantHandler
+     *
+     * @throws IdentityOAuth2Exception Error when initializing
+     */
+    public void init() throws IdentityOAuth2Exception {
+
+        super.init();
+
+        String validateIATProp = IdentityUtil.getProperty(TokenExchangeConstants.PROP_ENABLE_IAT_VALIDATION);
+        validateIAT = Boolean.parseBoolean(validateIATProp);
+
+        String validityPeriodProp = IdentityUtil.getProperty(TokenExchangeConstants.PROP_IAT_VALIDITY_PERIOD);
+
+        if (validateIAT) {
+            if (StringUtils.isNotBlank(validityPeriodProp)) {
+                try {
+                    validityPeriod = Integer.parseInt(validityPeriodProp);
+                } catch (NumberFormatException e) {
+                    validityPeriod = TokenExchangeConstants.DEFAULT_IAT_VALIDITY_PERIOD;
+                    log.warn("Invalid value: " + validityPeriodProp + " is set for IAT validity period. Using " +
+                            "default value: " + validityPeriod + " minutes.");
+                }
+            } else {
+                validityPeriod = TokenExchangeConstants.DEFAULT_IAT_VALIDITY_PERIOD;
+                log.warn("Empty value is set for IAT validity period. Using default value: " + validityPeriod
+                        + " minutes.");
+            }
+        } else {
+            log.debug("IAT Validation is disabled for JWT");
+        }
+
+        String registeredClaims = IdentityUtil.getProperty(TokenExchangeConstants.REGISTERED_CLAIMS);
+        if (StringUtils.isNotBlank(registeredClaims)) {
+            registeredClaimNames = registeredClaims.split("\\s*,\\s*");
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Validate IAT is set to: " + validateIAT + " for JWT grant.");
+            if (validateIAT) {
+                log.debug("IAT validity period is set to: " + validityPeriod + " minutes for JWT grant.");
+            }
+        }
+    }
+
+    /**
+     * Validate the Token Exchange Grant.
+     * Checks whether the token request satisfies the requirements to exchange the token.
+     *
+     * @param tokReqMsgCtx OAuthTokenReqMessageContext
+     * @return true or false if the grant_type is valid or not.
+     * @throws IdentityOAuth2Exception Error when validating the Token Exchange Grant
+     */
+    @Override
+    public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2Exception {
+
+        SignedJWT signedJWT;
+        IdentityProvider identityProvider;
+        String tokenEndPointAlias;
+        JWTClaimsSet claimsSet = null;
+        String requested_audience = null;
+        boolean audienceFound;
+
+        RequestParameter[] params = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getRequestParameters();
+        Map<String, String[]> requestParams = Arrays.stream(params).collect(Collectors.toMap(RequestParameter::getKey,
```

Comment:
Is there a case where we would have multiple values for the same param in this grant type?
If not I would suggest to make this a Map<String, String> and add the 0th element of each key as the value

Reply:
Changed :)